src/filesystem/zip_folder_accessor.cpp

```diff
@@ -3,196 +3,206 @@
-                        cur_node = child.get();
+                        cur_node = &child;
 
                         node_found = true;
-                        break;
+
+                        return false;
                     }
-                }
-                if(node_found) {
-                    continue;
-                }
 
-                // We didn't find a node for the current part of the path, so let's add one
-                auto new_node = std::make_unique<FileTreeNode>();
-                new_node->name = part;
-                new_node->parent = cur_node;
+                    return true;
+                });
 
-                auto* new_node_raw = new_node.get();
-                cur_node->children.push_back(std::move(new_node));
+                if(!node_found) {
+                    // We didn't find a node for the current part of the path, so let's add one
+                    FileTreeNode new_node;
+                    new_node.name = part;
+                    new_node.parent = cur_node;
 
-                cur_node = new_node_raw;
-            }
-        }
+                    cur_node->children.push_back(new_node);
+
+                    cur_node = &cur_node->children.last();
+                }
+            });
+        });
     }
 
-    bool ZipFolderAccessor::does_resource_exist_on_filesystem(const fs::path& resource_path) {
-        const auto resource_string = resource_path.string();
-        const auto existence_maybe = does_resource_exist_in_map(resource_string);
+    bool ZipFolderAccessor::does_resource_exist_on_filesystem(const rx::string& resource_path) {
+        const auto existence_maybe = does_resource_exist_in_map(resource_path);
         if(existence_maybe) {
-            return existence_maybe.value();
+            return *existence_maybe;
         }
 
-        int32_t ret_val = mz_zip_reader_locate_file(&zip_archive, resource_string.c_str(), "", 0);
+        const int32_t ret_val = mz_zip_reader_locate_file(&zip_archive, resource_path.data(), "", 0);
         if(ret_val != -1) {
             // resource found!
-            resource_indexes.emplace(resource_string, ret_val);
-            resource_existence.emplace(resource_string, true);
+            resource_indexes.insert(resource_path, ret_val);
+            resource_existence.insert(resource_path, true);
             return true;
         }
+
         // resource not found
-        resource_existence.emplace(resource_string, false);
+        resource_existence.insert(resource_path, false);
         return false;
     }
 
-    void print_file_tree(const std::unique_ptr<FileTreeNode>& folder, const uint32_t depth) {
-        if(folder == nullptr) {
-            return;
-        }
-
+    void print_file_tree(const FileTreeNode& folder, const uint32_t depth) {
         std::stringstream ss;
         for(uint32_t i = 0; i < depth; i++) {
             ss << "    ";
         }
 
-        ss << folder->name;
+        ss << folder.name.data();
```

Comment:
Note for the future: this should be an overload you write so you don't have to do the whole data thing.

Reply:
I'll want some string builder class tbh. Not really sure what form it'll take, though

Reply:
Makes sense. I would say to go the rust/rex way with a format string.

Reply:
Only thing is I don't know how many thing I'll need to add to the format string ahead of time, so idk how long to make it? At some point I'll have to dynamically build a string and it'll be gross no matter what